internal/core/application/wallet_service.go

```diff
@@ -774,29 +787,28 @@ func (w *walletService) restoreUnspents(
 				"account %d index %d", in.AccountIndex, addressIndex,
 			),
 		}
-		go w.restoreUnspentsForAddress(cb, in, chUnspentsInfo, chErr)
+		go w.restoreUnspentsForAddress(cb, in, chUnspentsInfo, wg)
+		time.Sleep(1 * time.Millisecond)
+	}
```

Comment:
go func() {
   defer close(chUnspentsInfo)
   for i := range info {
		in := info[i]
		path := strings.Split(in.DerivationPath, "/")
		addressIndex, _ := strconv.Atoi(path[len(path)-1])
		chRes <- &InitWalletReply{
			AccountIndex: in.AccountIndex,
			AddressIndex: addressIndex,
			Status:       Processing,
			Data: fmt.Sprintf(
				"account %d index %d", in.AccountIndex, addressIndex,
			),
		}
		go w.restoreUnspentsForAddress(cb, in, chUnspentsInfo, wg)
		time.Sleep(1 * time.Millisecond)
	}
        wg.Wait()
}